redash/handlers/query_results.py

```diff
@@ -132,6 +132,31 @@ def post(self):
 ONE_YEAR = 60 * 60 * 24 * 365.25
 
 
+class QueryResultSetResource(BaseResource):
+    @require_permission('view_query')
+    def get(self, query_id=None, filetype='json'):
+        query = get_object_or_404(models.Query.get_by_id_and_org, query_id, self.current_org)
+        if not query.schedule_keep_results:
+            abort(404, message="query does not keep multiple results")
+
+        # Synthesize a result set from the last N results.
+        total = len(query.query_results)
+        offset = max(total - query.schedule_keep_results, 0)
+        results = [qr.to_dict() for qr in query.query_results[offset:offset + total]]
+        if not results:
+            aggregate_result = {}
+        else:
+            aggregate_result = results[0].copy()
```

Comment:
I'm not sure why this picks up the first result only here?

Reply:
the code below copies the rest of the result sets into its `rows`.